app/actions/AuthenticatedAction.scala

```diff
@@ -93,19 +88,76 @@ class AuthenticatedAction @Inject()(
     }
   }
 
-  private def handleResult(result: AuthorisationResult, body: BasicAuthenticatedRequest[AnyContent] => Future[Result])
-                          (implicit request: Request[AnyContent]) = {
+  def success[A](
+                        accounts: Accounts,
+                        body: BasicAuthenticatedRequest[A] => Future[Result])
+                      (implicit request: Request[A], hc: HeaderCarrier): Future[Result] = {
+    def handleError: PartialFunction[Throwable, Future[Result]] = {
+      case _: NoActiveSession =>
+        provider.redirectToLogin
+      case otherException =>
+        Logger.debug(s"Exception thrown on authorisation with message: ${otherException.getMessage}")
```

Comment:
can we log the exception, the full stack trace as well.

Reply:
removed